src/renderer/containers/AddValidator/CheckPassword.tsx

```diff
@@ -38,27 +34,20 @@ Pick<IRootState, "register" | "addValidator" | "auth">> = (props) => {
         if(accounts!=null){
             const isCorrectValue = await accounts.isCorrectPassword(input);
             if(isCorrectValue) {
-                
+
                 const signingKey = PrivateKey.fromBytes(
-                    Buffer.from(props.register.signingKey.slice(2), "hex")
-                );
-                
-                if(props.auth.auth !== null) {
+                    Buffer.from(props.register.signingKey.slice(2), "hex"));
+
+                if(props.auth.auth !== null)
                     props.auth.auth.addValidator(new Keypair(signingKey));
-                    const accountDirectory = path.join(getConfig(remote.app).storage.accountsDir, DEFAULT_ACCOUNT);
-                    await V4Keystore.create(
-                        path.join(accountDirectory, PublicKey.fromPrivateKey(signingKey).toHexString() + ".json"),
-                        input, new Keypair(signingKey)
-                    );
-                    
-                    await database.account.set(
-                        DEFAULT_ACCOUNT, 
-                        props.auth.auth
-                    );
-                }
+                
+                props.afterPassword(input);
+                
                 setInputStatus(true);
-                props.storeAddValidator(false);
-                setTimeout(props.history.replace,500,Routes.DASHBOARD_ROUTE);
+                setTimeout(props.history.push,500,{
```

Comment:
Can you remind me why we have this delay?

Reply:
To briefly see green border and checkmark of Input component, after correct submit.